geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapPut.java

```diff
@@ -396,12 +410,35 @@ private boolean checkUninitializedRegionPreconditions() {
   private boolean checkCreatePreconditions() {
     if (isIfNew()) {
       if (!getRegionEntry().isDestroyedOrRemoved()) {
+        // retain the version stamp of the existing entry for use in processing failures
+        EntryEventImpl event = getEvent();
+        if (event.getRegion().getConcurrencyChecksEnabled() &&
+            event.getOperation() == Operation.PUT_IF_ABSENT &&
+            !event.hasValidVersionTag() &&
+            event.isPossibleDuplicate()) {
+          if (ValueComparisonHelper.checkEquals(getRegionEntry().getValue(),
+              getEvent().basicGetNewValue(),
```

Comment:
I think getRawNewValue is better than basicGetNewValue. They currently are the same thing but I think the "basic*" methods should not be public.